### PR TITLE
Moved the get_price responsibility from DataProvider to AbstractPriceDataProvider

### PR DESCRIPTION
**Pull request by myrmarachne**
- Moved get_price and other pricing functionality from DataProvider to AbstractPriceDataProvider
- Moved all futures functionality to a new abstract class FuturesDataProvider. All classes returning futures data should now extend that class.
- Removed DataHandler, instead DataProviders have a timer object. Backtests should use SettableTimer, while real live trading should use RealTimer (default one). In case of SettableTimer, if looking into the future would be desired - look_ahead_bias flag has been added to get_history / get_price. By default it is set to False, but overriding it will allow taking a future value within a backtest.

### CODE REVIEW

File: qf_lib/tests/integration_tests/backtesting/data_handler/test_data_handler.py [hunk 1]
```diff
@@ -32,7 +32,7 @@
 from qf_lib.tests.integration_tests.connect_to_data_provider import get_data_provider
 
 
-class TestDataHandler(TestCase):
```
Comment (marekbais) [inline qf_lib/tests/integration_tests/backtesting/data_handler/test_data_handler.py:35]: let's also rename the file and delete data_handler dir

File: qf_lib/backtesting/execution_handler/market_orders_executor.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def _get_current_prices(self, tickers: Sequence[Ticker]):
                                                    " executor"
```
Comment (marekbais) [inline qf_lib/backtesting/execution_handler/market_orders_executor.py:81]: remove reference to data handler from docstring in line 74

File: qf_lib/backtesting/trading_session/backtest_trading_session.py [hunk 3]
```diff
@@ -41,10 +41,10 @@ class BacktestTradingSession(TradingSession):
...
-                 timer: SettableTimer, notifiers: Notifiers, portfolio: Portfolio, events_manager: EventManager,
-                 monitor: BacktestMonitor, broker: BacktestBroker, order_factory: OrderFactory, frequency: Frequency,
-                 backtest_result: BacktestResult):
+                 position_sizer: PositionSizer, orders_filters: Sequence[OrdersFilter],
+                 data_provider: AbstractPriceDataProvider, notifiers: Notifiers,
+                 portfolio: Portfolio, events_manager: EventManager, monitor: BacktestMonitor, broker: BacktestBroker,
+                 order_factory: OrderFactory, frequency: Frequency, backtest_result: BacktestResult):
         """
         Set up the backtest variables according to what has been passed in.
         The data_provider parameter of the BacktestTradingSession points to a Data Handler object.
```
Comment (marekbais) [inline qf_lib/backtesting/trading_session/backtest_trading_session.py:50]: remove reference to data handler

File: qf_lib/tests/integration_tests/data_providers/bloomberg/test_bbg_data_handler.py [hunk 4]
```diff
@@ -25,14 +24,13 @@
 from qf_lib.tests.integration_tests.connect_to_data_provider import get_data_provider
 
 
-class TestBloombergDataHandler(TestCase):
+class TestBloombergDataProvider(TestCase):
     """Class which tests Data Handler encapsulation with Bloomberg Data Provider."""
```
Comment (marekbais) [inline qf_lib/tests/integration_tests/data_providers/bloomberg/test_bbg_data_handler.py:28]: remove data handler for docstring and filename

File: qf_lib/containers/futures/future_tickers/future_ticker.py [hunk 5]
```diff
@@ -87,16 +87,17 @@ def __init__(self, name: str, family_id: str, N: int, days_before_exp_date: int,
         self._last_cached_datetime = None
         self._expiration_hour = RelativeDelta(hour=0, minute=0, second=0, microsecond=0)
 
-    def initialize_data_provider(self, timer: Timer, data_provider: "DataProvider"):
+    def initialize_data_provider(self, timer: Timer, data_provider: "FuturesDataProvider"):
```
Comment (eirik-thorp) [inline qf_lib/containers/futures/future_tickers/future_ticker.py:90]: For this part, the futures ticker has a timer and a data_provider, but shouldn't now the data_provider have the timer and the ticker use that timer?

**Review by zeynepgultugaydemir — APPROVED** [review]
LGTM 👍

**Review by tavisit — APPROVED** [review]

File: .coverargerc [hunk 6]
```diff
@@ -18,6 +18,7 @@ exclude_lines =
     # Don't complain about missing debug-only code:
     def __repr__
     def __str__
+    pass
```
Comment (eirik-thorp) [inline .coverargerc:21]: No

**Review by eirik-thorp — DISMISSED** [review]
Wait

**Review by tavisit — APPROVED** [review]

**Review by marekbais — APPROVED** [review]

**Pull request merged**